addons/web_linkedin/models/web_linkedin.py

```diff
@@ -0,0 +1,428 @@
+        warnings = []
+        if not params:
+            params = {'oauth2_access_token': self.env.user.linkedin_token}
+        if not headers:
+            headers = {'Content-type': 'application/json', 'Accept': 'text/plain', 'x-li-format': 'json'}
+        if kw.get('search_uid'):
+            #this code may returns 400 bad request error, as per linked API doc the call is proper
+            #but generated url may not have proper public url and may raise 400 or 410 status hence added a warning in response and handle warning at client side
+            try:
+                public_profile_url = werkzeug.url_quote_plus("http://www.linkedin.com/pub/%s" % (kw['search_uid']))
+                profile_uri = "/people/url={public_profile_url}:{people_fields}".format(public_profile_url=public_profile_url, people_fields='(' + ','.join(PEOPLE_FIELDS) + ')')
+                status, public_profile = self.with_context(kw.get('local_context') or {}).send_request(profile_uri, params=params, headers=headers, type="GET")
+
+            except urllib2.HTTPError, e:
+                if e.code in (400, 410):
+                    warnings.append([_('LinkedIn error'), _('LinkedIn is temporary down for the searches by url.')])
+                elif e.code in (401):
+                    raise e
+        search_params = dict(params.copy(), keywords=kw.get('search_term', "") or "", start=offset, count=limit)
+        #Note: People search is allowed to only vetted API access request, please go through following link
+        #https://help.linkedin.com/app/api-dvr
+        people_search_uri = "/people-search:(people:{people_fields})".format(people_fields='(' + ','.join(PEOPLE_FIELDS) + ')')
+        status, people = self.with_context(kw.get('local_context') or {}).send_request(people_search_uri, params=search_params, headers=headers, type="GET")
+        if people and people['people'].get('values') and public_profile:
+            people['people']['values'].append(public_profile)
+        return status, people, warnings
+
+    @api.model
+    def get_people_from_company(self, company_universalname, limit, from_url):
+        if not self.need_authorization():
+            headers = {'Content-type': 'application/json', 'Accept': 'text/plain', 'x-li-format': 'json'}
+            params = {
+                'oauth2_access_token': self.env.user.linkedin_token,
+                'company-name': company_universalname,
+                'current-company': 'true',
+                'count': limit
+            }
+            people_criteria_uri = "/people-search:(people:{people_fields})".format(people_fields='(' + ','.join(PEOPLE_FIELDS) + ')')
+            status, res = self.send_request(people_criteria_uri, params=params, headers=headers, type="GET")
+            return res
+        else:
+            return {'status': 'need_auth', 'url': self._get_authorize_uri(from_url=from_url)}
+
+    def send_request(self, uri, pre_uri=BASE_API_URL, params={}, headers={}, type="GET"):
```

Comment:
There is no existing method doing this ? If not, that should probably go in tools.